## Cifar10

In [2]:
from tensorflow.python.keras.datasets import cifar10
(x_train,y_train),(x_test,y_test) = cifar10.load_data()

In [3]:
from tensorflow.python.client import device_lib
device_lib.list_local_devices()

[name: "/device:CPU:0"
 device_type: "CPU"
 memory_limit: 268435456
 locality {
 }
 incarnation: 449364820353077048, name: "/device:XLA_CPU:0"
 device_type: "XLA_CPU"
 memory_limit: 17179869184
 locality {
 }
 incarnation: 14801721957984933094
 physical_device_desc: "device: XLA_CPU device"]

In [4]:
print("x_train.shape: ",x_train.shape)
print("x_test.shapa: ",x_test.shape)
print("y_train.shape: ",y_train.shape)
print("y_test.shape: ",y_test.shape)

x_train.shape:  (50000, 32, 32, 3)
x_test.shapa:  (10000, 32, 32, 3)
y_train.shape:  (50000, 1)
y_test.shape:  (10000, 1)


In [5]:
from tensorflow.python.keras.utils import to_categorical

x_train = x_train/255
x_test = x_test/255
y_train = to_categorical(y_train,10)
y_test = to_categorical(y_test,10)

In [6]:
from tensorflow.python.keras.models import Sequential
from tensorflow.python.keras.layers import Conv2D
from tensorflow.python.keras.layers import MaxPooling2D
from tensorflow.python.keras.layers import Dropout
from tensorflow.python.keras.layers import Dense
from tensorflow.python.keras.layers import Flatten
from tensorflow.python.keras.callbacks import TensorBoard


In [7]:
import tensorflow as tf

In [8]:
strategy = tf.distribute.MirroredStrategy()

In [14]:
model = Sequential()

model.add(
    Conv2D(
        filters=32,
        input_shape=(32,32,3),
        kernel_size=(3,3),
        strides=(1,1),
        padding='same',
        activation="relu"
    )
)

model.add(
    Conv2D(
        filters=32,
        kernel_size=(3,3),
        strides=(1,1),
        padding='same',
        activation="relu"
    )
)
model.add(MaxPooling2D(pool_size=(2,2)))
model.add(Dropout(0.25))

model.add(
    Conv2D(
        filters=64,
        kernel_size=(3,3),
        strides=(1,1),
        activation='relu'
    )
)

model.add(
    Conv2D(
        filters=64,
        kernel_size=(3,3),
        strides=(1,1),
        padding="same",
        activation="relu"
    )
)

model.add(MaxPooling2D(pool_size=(2,2)))
model.add(Dropout(0.25))

model.add(Flatten())
model.add(Dense(units=512,activation="relu"))
model.add(Dropout(0.5))
model.add(Dense(units=10,activation="softmax"))

model.compile(
optimizer="adam",
loss="categorical_crossentropy",
metrics=['accuracy']
)

tsb = TensorBoard(log_dir="./log")
history_model1 = model.fit(
    x_train,
    y_train,
    batch_size=32,
    epochs=2,
    validation_split=0.2,
    callbacks=[tsb]
)

Train on 40000 samples, validate on 10000 samples
Epoch 1/2
40000/40000 [==============================] - 59s 1ms/sample - loss: 1.5642 - acc: 0.4308 - val_loss: 1.1570 - val_acc: 0.5818
Epoch 2/2
40000/40000 [==============================] - 44s 1ms/sample - loss: 1.1450 - acc: 0.5936 - val_loss: 0.9565 - val_acc: 0.6623


In [13]:
with strategy.scope():
    model = Sequential()

    model.add(
        Conv2D(
            filters=32,
            input_shape=(32,32,3),
            kernel_size=(3,3),
            strides=(1,1),
            padding='same',
            activation="relu"
        )
    )

    model.add(
        Conv2D(
            filters=32,
            kernel_size=(3,3),
            strides=(1,1),
            padding='same',
            activation="relu"
        )
    )
    model.add(MaxPooling2D(pool_size=(2,2)))
    model.add(Dropout(0.25))

    model.add(
        Conv2D(
            filters=64,
            kernel_size=(3,3),
            strides=(1,1),
            activation='relu'
        )
    )

    model.add(
        Conv2D(
            filters=64,
            kernel_size=(3,3),
            strides=(1,1),
            padding="same",
            activation="relu"
        )
    )

    model.add(MaxPooling2D(pool_size=(2,2)))
    model.add(Dropout(0.25))
    
    model.add(Flatten())
    model.add(Dense(units=512,activation="relu"))
    model.add(Dropout(0.5))
    model.add(Dense(units=10,activation="softmax"))
    
    model.compile(
    optimizer="adam",
    loss="categorical_crossentropy",
    metrics=['accuracy']
    )

    tsb = TensorBoard(log_dir="./log")
    history_model1 = model.fit(
        x_train,
        y_train,
        batch_size=32,
        epochs=2,
        validation_split=0.2,
        callbacks=[tsb]
    )

KeyboardInterrupt: 

Train on 40000 samples, validate on 10000 samples  
Epoch 1/2  
40000/40000 [==============================] - 121s 3ms/sample - loss: 1.5324 - accuracy: 0.4431 - val_loss: 1.2120 - val_accuracy: 0.5660  
Epoch 2/2  
40000/40000 [==============================] - 120s 3ms/sample - loss: 1.1634 - accuracy: 0.5842 - val_loss: 0.9441 - val_accuracy: 0.6662


Epoch 1/2  
40000/40000 [==============================] - 121s 3ms/sample - loss: 1.5324 - accuracy: 0.4431 - val_loss: 1.2120 - val_accuracy: 0.5660  
Epoch 2/2  
40000/40000 [==============================] - 120s 3ms/sample - loss: 1.1634 - accuracy: 0.5842 - val_loss: 0.9441 - val_accuracy: 0.6662  

poch 1/2  
40000/40000 [==============================] - 59s 1ms/sample - loss: 1.5642 - acc: 0.4308 - val_loss: 1.1570 - val_acc: 0.5818  
Epoch 2/2  
40000/40000 [==============================] - 44s 1ms/sample - loss: 1.1450 - acc: 0.5936 - val_loss: 0.9565 - val_acc: 0.6623  